lib/elixir/lib/kernel.ex

```diff
@@ -3271,7 +3271,12 @@ defmodule Kernel do
       if Map.has_key?(fields, :message) do
         @spec message(Exception.t) :: String.t
         def message(exception) do
-          exception.message
+          cond do
+            is_nil(exception.message) ->
```

Comment:
yes, i also agree with both of you, my question is what the is final output we want.
I am raising now:
            is_nil(exception.message) ->
              raise ArgumentError, "argument error: #{inspect(exception.message)}"

which gets printed as 
** (OrigininalExceptionError) got ArgumentError with message `argument error: nil` while retrieving Exception.message/1 for %TestEmptyError{message: nil}"

would that suffice?Or we need to reraise?